src/domain/entities/Questionnaire/Questionnaire.ts

```diff
@@ -272,10 +272,15 @@ export class Questionnaire {
             ...allQsInQuestionnaireStages,
         ];
 
+        const allQsInQuestionnaireWithUpdatedQ = allQsInQuestionnaire.map(question => {
+            if (question.id === updatedQuestion.id) return updatedQuestion;
+            else return question;
```

Comment:
This or a ternary? when it's a short one-line expression it seems nicer, but that's subjective.